lib/lightstep/instrumentation.rb

```diff
@@ -0,0 +1,41 @@
+module LightStep
+  # Instrument an operation. This is may be called by any LightStep component to
+  # record an instrumentation event. This is a compatible API to
+  # ActiveSupport::Notifications.
+  #
+  # event   - Name of the event as a symbol. The event is published under the
+  #           lightstep namespace automatically.
+  # payload - Hash payload for the event. Available keys are based on the event
+  #           being published.
+  #
+  # Returns the result of executing the block.
+  def self.instrument(event, payload = {}, &block)
+    if @instrumenter
+      block ||= proc {}
+      @instrumenter.instrument("#{event}.lightstep", payload, &block)
```

Comment:
They payload here can be very large, and the subscriber can keep it on the heap forever. Is it feasible to send a smaller payload to the subscriber that will still let them know when an exception has occurred?

Perhaps we can pass information about the report_request itself without passing the actual list of spans.